Framework/API/src/Algorithm.cpp

```diff
@@ -114,24 +115,37 @@ Algorithm::~Algorithm() {}
 //===================================
 //=============================================================================================
 
+/// Gets the current execution state
+ExecutionState Algorithm::executionState() const { return m_executionState; }
+
+/// Sets the current execution state
+void Algorithm::setExecutionState(const ExecutionState state) {
+  m_executionState = state;
```

Comment:
Hmm.  I see what you are getting at, but in some ways this feels like a side effect.  This is not a solid state model in that it is possible to externally move back a state from running to say initialised, but you should not really do that, and the base algorithm class does not do that.
I left the methods as protected in case someone in a specialized algorithm needs to control something in more detail, but I can't think of an example at the moment that would need it.
I feel if you are fiddling with the state within your algorithm expecting you to set the ResultState as well is not too much to ask.